csharp/src/Ice/IncomingFrame.cs

```diff
@@ -65,6 +65,10 @@ public abstract class IncomingFrame
         /// <summary>The frame data.</summary>
         internal ArraySegment<byte> Data { get; set; }
 
+        /// <summary>The socket stream used to receive data for input stream parameters. It's only assigned if the
+        /// socket stream wasn't finished after reading this frame.</summary>
+        internal SocketStream? SocketStream { get; set; }
```

Comment:
I think we can move this property to the derived classes and make it get-only.

Reply:
It's now a private readonly data member.